src/main/java/me/shedaniel/fiber2cloth/api/Fiber2Cloth.java

```diff
@@ -29,34 +41,30 @@ static Fiber2Cloth create(Screen parentScreen, String modId, ConfigNode node, St
     
     Fiber2Cloth setAfterInitConsumer(Consumer<Screen> afterInitConsumer);
     
-    Node getDefaultCategoryNode();
+    ConfigBranch getDefaultCategoryNode();
     
-    Fiber2Cloth setDefaultCategoryNode(Node defaultCategoryNode);
+    Fiber2Cloth setDefaultCategoryNode(ConfigBranch defaultCategoryNode);
     
     Fiber2Cloth setSaveRunnable(Runnable saveRunnable);
-    
-    default Fiber2Cloth registerNodeEntryFunction(Node node, Function function) {
-        registerTreeEntryFunction(node, function);
-        return this;
-    }
-    
-    default Fiber2Cloth hideNode(Node node) {
+
+    default Fiber2Cloth hideNode(ConfigBranch node) {
         hideTreeEntry(node);
         return this;
     }
     
-    default Fiber2Cloth hideTreeEntry(TreeItem item) {
+    default Fiber2Cloth hideTreeEntry(ConfigNode item) {
         registerTreeEntryFunction(item, null);
         return this;
     }
     
-    Fiber2Cloth registerTreeEntryFunction(TreeItem item, Function function);
-    
-    Fiber2Cloth registerNodeEntryFunction(Class clazz, Function<ConfigValue, Object> function);
-    
-    Map<Class, Function> getFunctionMap();
-    
-    ConfigNode getNode();
+    Fiber2Cloth registerTreeEntryFunction(ConfigNode item, Function<ConfigNode, AbstractConfigListEntry<?>> function);
```

Comment:
If it is not too much, I would like this to return a list of config entry so one more can create multiple entries to behave like auto config.